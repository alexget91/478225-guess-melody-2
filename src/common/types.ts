import {QuestionType} from "./constants";

type Song = {
  artist: string,
  picture: string,
};

type ArtistAnswer = {
  artist: string,
  picture: string,
};

type GenreAnswer = {
  genre: string,
  src: string,
};

type ArtistQuestion = {
  type: QuestionType,
  song: Song,
  answers: Array<ArtistAnswer>,
};

type GenreQuestion = {
  type: QuestionType,
  genre: string,
  answers: Array<GenreAnswer>,
};

type User = {
  id: number,
  email: string,
};

export {Song, ArtistQuestion, GenreQuestion, User};
