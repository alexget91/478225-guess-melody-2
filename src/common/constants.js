const ScreenSteps = {
  START: -1,
  FAIL_TIME: -2,
  FAIL_MISTAKES: -3,
};

export {ScreenSteps};
