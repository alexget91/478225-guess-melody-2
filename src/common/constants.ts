enum ScreenStep {
  START = -1,
}

enum QuestionType {
  ARTIST = `artist`,
  GENRE = `genre`,
}

enum FailType {
  TIME = `TIME`,
  MISTAKES = `MISTAKES`,
}

export {ScreenStep, QuestionType, FailType};
