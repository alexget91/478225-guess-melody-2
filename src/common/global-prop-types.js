import PropTypes from "prop-types";

const genreQuestion = {
  type: PropTypes.oneOf([`genre`, `artist`]).isRequired,
  genre: PropTypes.string.isRequired,
  answers: PropTypes.arrayOf(PropTypes.exact({
    genre: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  })).isRequired,
};

const artistQuestion = {
  type: PropTypes.oneOf([`genre`, `artist`]).isRequired,
  song: PropTypes.exact({
    artist: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  }).isRequired,
  answers: PropTypes.arrayOf(PropTypes.exact({
    artist: PropTypes.string.isRequired,
    picture: PropTypes.string.isRequired,
  })).isRequired
};

const userData = {
  id: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
};

export {genreQuestion, artistQuestion, userData};
