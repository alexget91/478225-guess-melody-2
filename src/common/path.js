const Path = {
  INDEX: `/`,
  AUTH: `/auth`,
  WIN: `/win`,
  LOSE_MISTAKES: `/lose-mistakes`,
  LOSE_TIME: `/lose-time`,
};

export default Path;
