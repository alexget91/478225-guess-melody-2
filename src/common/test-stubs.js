const createAudioMock = () => {
  return {createNodeMock(element) {
    if (element.type === `audio`) {
      return {
        createRef() {}
      };
    }
    return null;
  }};
};

export default createAudioMock;
