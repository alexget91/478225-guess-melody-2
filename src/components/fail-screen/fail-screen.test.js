import React from "react";
import renderer from "react-test-renderer";
import FailScreen from "../fail-screen/fail-screen";
import {FailType} from "../../common/constants";
import {MemoryRouter} from "react-router-dom";

describe(`Fail screen correctly renders after relaunch`, () => {
  it(`Time fail`, () => {
    const tree = renderer
      .create(<MemoryRouter>
        <FailScreen failType={FailType.TIME} onReplayClick={jest.fn()}/>
      </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Mistakes fail`, () => {
    const tree = renderer
      .create(<MemoryRouter>
        <FailScreen failType={FailType.MISTAKES} onReplayClick={jest.fn()}/>
      </MemoryRouter>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
