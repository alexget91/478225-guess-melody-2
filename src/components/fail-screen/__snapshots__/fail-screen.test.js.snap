// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fail screen correctly renders after relaunch Mistakes fail 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Какая жалость!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    У вас закончились все попытки. Ничего, повезёт в следующий раз!
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Попробовать ещё раз
  </a>
</section>
`;

exports[`Fail screen correctly renders after relaunch Time fail 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Увы и ах!
  </h2>
  <p
    className="result__total result__total--fail"
  >
    Время вышло! Вы не успели отгадать все мелодии
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Попробовать ещё раз
  </a>
</section>
`;
